template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseCloudStorageServiceConfigProviderTest.java

```diff
@@ -150,6 +152,43 @@ public void testConfigurationNeededWhenDatalake728() {
         assertTrue(configurationNeeded);
     }
 
+    @Test
+    public void testConfigurationNeededWhenDatalake727AWSNoEntitlement() {
+        when(entitlementService.sdxHbaseCloudStorageEnabled(anyString())).thenReturn(false);
+
+        TemplatePreparationObject preparationObject = getTemplatePreparationObject(true, true, "7.2.7", CloudPlatform.AWS);
+        String inputJson = getBlueprintText("input/clouderamanager.bp");
+        CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(inputJson);
+
+        boolean configurationNeeded = underTest.isConfigurationNeeded(cmTemplateProcessor, preparationObject);
+        assertTrue(configurationNeeded);
```

Comment:
as @lajosrodek wrote, it is hard to understand `HbaseCloudStorageServiceConfigProvider#isConfigurationNeeded` method and the complexity of it is high, also the code of this test method is similar to the other test methods, I suggest to rewrite these test methods to one parameterized unit test to cover all cases (based on the complexity of the return statement of `HbaseCloudStorageServiceConfigProvider#isConfigurationNeeded`, there are many cases), here is an example: https://github.com/hortonworks/cloudbreak/blob/9e5bf09c73c6f25c8a2da5fb88567da2c64b3b33/core/src/test/java/com/sequenceiq/cloudbreak/service/ServiceEndpointCollectorVersionComparatorTest.java but you can find lot of examples in the Cloudbreak code for `@ParameterizedTest`

Reply:
+1 for using `@ParameterizedTest`, but this requires migrating the class to JUnit 5 as I suggested above.

Reply:
I have updated to use parameterized test. It is much cleaner. Thanks!

Reply:
thank you!